src/plugins/data/public/search/session/sessions_mgmt/components/table/app_filter.tsx

```diff
@@ -18,12 +18,7 @@ export const getAppFilter: (tableData: UISession[]) => SearchFilterConfig = (tab
   }),
   field: 'appId',
   multiSelect: 'or',
-  options: tableData.reduce((options: FieldValueOptionType[], { appId }) => {
-    const existingOption = options.find((o) => o.value === appId);
-    if (!existingOption) {
-      return [...options, { value: appId, view: capitalize(appId) }];
-    }
-
-    return options;
-  }, []),
+  options: [...new Set(tableData.map((data) => data.appId ?? 'unknown'))]
+    .sort()
```

Comment:
After fixing sort order for the `find` some UI session tests started failing. They failed because the order of this option has changed. I fixed it by starting sorting the options to make them consistent. Also slightly refactored the function